src/vtarget-out/vtarget_gen_yosys.cc

```diff
@@ -0,0 +1,447 @@
+  vlg_wrapper.add_init_stmt(varname + " <= " + rand_in_name + ";");
+  vlg_wrapper.add_always_stmt(varname + " <= " + varname + ";");
+  add_an_assumption(
+      "(~(" + cond + ") || ((" + varname + ") == (" + expression + ")))", dspt);
+}
+
+/// Add an assumption
+void VlgSglTgtGen_Yosys::add_an_assumption(const std::string& aspt,
+                                          const std::string& dspt) {
+  auto assumption_wire_name = vlg_wrapper.sanitizeName(dspt) + new_mapping_id();
+  vlg_wrapper.add_wire(assumption_wire_name, 1, true);
+  vlg_wrapper.add_output(assumption_wire_name,
+                         1); // I find it is necessary to connect to the output
+  vlg_wrapper.add_assign_stmt(assumption_wire_name, aspt);
+  ILA_ERROR_IF(aspt.find(".") != std::string::npos)
+      << "-------- aspt:" << aspt << " contains unfriendly dot.";
+  if (dspt == "invariant_assume") {
+    ILA_WARN << "Ignore invariant assume : " << assumption_wire_name;
+    return;
+  }
+  _problems.assumptions[dspt].exprs.push_back(assumption_wire_name);
+  //_problems.assumptions.push_back(convert_expr_to_yosys(aspt));
+}
+/// Add an assertion
+void VlgSglTgtGen_Yosys::add_an_assertion(const std::string& asst,
+                                         const std::string& dspt) {
+  auto assrt_wire_name = vlg_wrapper.sanitizeName(dspt) + new_property_id();
+  vlg_wrapper.add_wire(assrt_wire_name, 1, true);
+  vlg_wrapper.add_output(assrt_wire_name,
+                         1); // I find it is necessary to connect to the output
+  vlg_wrapper.add_assign_stmt(assrt_wire_name, asst);
+  _problems.assertions[dspt].exprs.push_back(assrt_wire_name);
+  ILA_ERROR_IF(asst.find(".") != std::string::npos)
+      << "-------- asst:" << asst << " contains unfriendly dot.";
+  //_problems.probitem[dspt].assertions.push_back(convert_expr_to_yosys(asst));
+}
+
+/// Add an assumption
+void VlgSglTgtGen_Yosys::add_a_direct_assumption(const std::string& aspt,
+                                                const std::string& dspt) {
+  if (dspt == "invariant_assume") {
+    ILA_WARN << "Ignore invariant assume : " << aspt;
+    return;
+  }
+  _problems.assumptions[dspt].exprs.push_back(aspt);
+}
+/// Add an assertion
+void VlgSglTgtGen_Yosys::add_a_direct_assertion(const std::string& asst,
+                                               const std::string& dspt) {
+  _problems.assertions[dspt].exprs.push_back(asst);
+}
+
+/// Pre export work : add assume and asssert to the top level
+/// collect the type of assumptions and assertions
+/// Assertions: variable_map_assert (no invariant_assert --- not for invariant target)
+/// Assumptions:
+///      1. global: noreset, funcmap, absmem, additonal_mapping_control_assume
+///      2. only at starting state:
+///          variable_map_assume, issue_decode, issue_valid, ...
+///      Question: invariant assume? where to put it? Depends on ?
+
+void VlgSglTgtGen_Yosys::PreExportProcess() {
+  // .. ??
+
+  std::string global_assume = "`true";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'global_assume' is assigned a value that is never used.](https://app.codacy.com/app/Bo-Yuan-Huang/ILAng/pullRequest?prid=4555809)